kernel/src/table_changes/mod.rs

```diff
@@ -126,14 +129,22 @@ impl TableChanges {
     pub fn table_root(&self) -> &Url {
         &self.table_root
     }
-    #[allow(unused)]
     pub(crate) fn partition_columns(&self) -> &Vec<String> {
         &self.end_snapshot.metadata().partition_columns
     }
-    #[allow(unused)]
     pub(crate) fn column_mapping_mode(&self) -> &ColumnMappingMode {
         &self.end_snapshot.column_mapping_mode
     }
+
+    /// Create a [`TableChangesScanBuilder`] for an `Arc<TableChanges>`.
+    pub fn scan_builder(self: Arc<Self>) -> TableChangesScanBuilder {
+        TableChangesScanBuilder::new(self)
+    }
+
+    /// Consume this `TableChanges` to create a [`TableChangesScanBuilder`]
+    pub fn into_scan_builder(self) -> TableChangesScanBuilder {
+        TableChangesScanBuilder::new(self)
+    }
```

Comment:
how about we only do the `scan_builder` API? unless we think we typically won't have an Arc laying around? if that's the case then only do this one?
```suggestion
```

Reply:
This is the same api  that `Snapshot` has, so I thought we could keep it consistent.